metrics/prometheus.go

```diff
@@ -0,0 +1,348 @@
+package metrics
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	info "github.com/google/cadvisor/info/v1"
+	"github.com/google/cadvisor/manager"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type prometheusMetric struct {
+	valueType prometheus.ValueType
+	value     float64
+	labels    []string
+}
+
+// PrometheusCollector implements prometheus.Collector.
+type PrometheusCollector struct {
+	manager manager.Manager
+
+	errors   prometheus.Gauge
+	lastSeen *prometheus.Desc
+
+	cpuUsageSeconds       *prometheus.Desc
+	cpuUsageSecondsPerCPU *prometheus.Desc
+
+	memoryUsageBytes *prometheus.Desc
+	memoryWorkingSet *prometheus.Desc
+	memoryFailures   *prometheus.Desc
+
+	fsLimit        *prometheus.Desc
+	fsUsage        *prometheus.Desc
+	fsReads        *prometheus.Desc
+	fsReadsSectors *prometheus.Desc
+	fsReadsMerged  *prometheus.Desc
+	fsReadTime     *prometheus.Desc
+
+	fsWrites        *prometheus.Desc
+	fsWritesSectors *prometheus.Desc
+	fsWritesMerged  *prometheus.Desc
+	fsWriteTime     *prometheus.Desc
+
+	fsIoInProgress *prometheus.Desc
+	fsIoTime       *prometheus.Desc
+
+	fsWeightedIoTime *prometheus.Desc
+
+	networkRxBytes   *prometheus.Desc
+	networkRxPackets *prometheus.Desc
+	networkRxErrors  *prometheus.Desc
+	networkRxDropped *prometheus.Desc
+	networkTxBytes   *prometheus.Desc
+	networkTxPackets *prometheus.Desc
+	networkTxErrors  *prometheus.Desc
+	networkTxDropped *prometheus.Desc
+
+	tasks *prometheus.Desc
+
+	descs []*prometheus.Desc
+}
+
+// NewPrometheusCollector returns a new PrometheusCollector.
+func NewPrometheusCollector(manager manager.Manager) *PrometheusCollector {
+	c := &PrometheusCollector{
+		manager: manager,
+		errors: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: "container",
+			Name:      "scrape_error",
+			Help:      "1 if there was an error while getting container metrics, 0 otherwise",
+		}),
+		lastSeen: prometheus.NewDesc(
+			"container_last_seen",
+			"Last time a container was seen by the exporter",
+			[]string{"name", "id"},
+			nil),
+		cpuUsageSeconds: prometheus.NewDesc(
+			"container_cpu_usage_seconds_total",
+			"Cumulative cpu time consumed in seconds.",
+			[]string{"name", "id", "type"},
+			nil),
+		cpuUsageSecondsPerCPU: prometheus.NewDesc(
+			"container_cpu_usage_seconds_total",
```

Comment:
That'll be confusing when aggregated up, i'd suggest container_cpu_usagetype_seconds_total for the other